src/cinder/audio/Target.cpp

```diff
@@ -81,7 +102,29 @@ void TargetFile::write( const Buffer *buffer, size_t numFrames, size_t frameOffs
 
 	CI_ASSERT_MSG( numFrames + frameOffset <= buffer->getNumFrames(), "numFrames + frameOffset out of bounds" );
 
-	performWrite( buffer, numFrames, frameOffset );
+	if( mConverter ) {
+		auto currFrame = frameOffset;
+		auto lastFrame = frameOffset + numFrames;
+
+		// process buffer in chunks of mMaxFramesPerConversion
+		while ( currFrame != lastFrame ) {
+			auto numSourceFrames = std::min( mMaxFramesPerConversion, lastFrame - currFrame );
+			auto numDestFrames = size_t( numSourceFrames * (float)getSampleRateTarget() / (float)getSampleRate() );
+
+			// copy buffer into temporary buffer to remove frame offset (needed for mConverter->convert)
+			mConverterSourceBuffer.copyOffset( *buffer, numSourceFrames, 0, currFrame );
```

Comment:
Concerning this and your comment in the PR description about this being a bit hacky, I hope that in the future this problem can be alleviated once `audio::BufferView`s are implemented (see https://github.com/cinder/Cinder/issues/1701). I think for now a TODO comment linking back to that would suffice. The copy is quite cheap compared to the actual encoding, but obviously eventually we want to do as little extra copying as possible.